src/main/java/net/dv8tion/jda/internal/requests/MemberChunkManager.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2015-2020 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.internal.requests;
+
+import gnu.trove.map.TLongObjectMap;
+import gnu.trove.map.hash.TLongObjectHashMap;
+import gnu.trove.set.TLongSet;
+import gnu.trove.set.hash.TLongHashSet;
+import net.dv8tion.jda.api.utils.MiscUtil;
+import net.dv8tion.jda.api.utils.data.DataObject;
+
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class MemberChunkManager
+{
+    private final WebSocketClient client;
+    private final ReentrantLock lock = new ReentrantLock();
+    private final TLongObjectMap<Queue<ChunkRequest>> requests = new TLongObjectHashMap<>();
+    private final TLongSet requestedChunk = new TLongHashSet();
```

Comment:
This implies it it a single chunk that is requested when really it is a set of Guilds that currently have outstanding chunk requests.
maybe `outstandingChunkRequests` or something similar would be a more clear naming?

Reply:
We will revisit this class later anyway since chunking gets some changes according to jake. This includes nonces which will allow us to make a map instead.

Reply:
Okay dope.

Reply:
Okay cool